syntax= "proto3";
import "google/protobuf/empty.proto";

// Services definitions
service GatewayService {
  rpc Identificate(Identification) returns (Id);
  rpc ReceiveStatus(Status) returns (Status);
}

service EquipmentService {
  rpc Identificate(google.protobuf.Empty) returns (Identification);
  rpc ReceiveUpdate(UpdateStatus) returns (google.protobuf.Empty);
  rpc GetValor(google.protobuf.Empty) returns (Status);
}

message Id {
  string value = 1;
}

enum EquipmentType {
  BOTH = 0;
  SENSOR = 1;
  ACTUATOR = 2;
}

// Messages definitions
message Identification {
  string name = 1;
  EquipmentType type = 2;
  string ip = 3;
  int32 port = 4;
}

// Lâmpada & sistema de irrigação ~ atuadores

message ActuatorStatus { 
  bool turnedOn = 4;
}

message TurnOnOff {
  bool turnedOn = 5;
}

// Ar-condicionado com sensor de temperatura ~ sensor

message AirConditioningStatus {
  bool turnedOn = 6;
  int32 temperature = 7;
  int32 roomTemperature = 8;
}

message Status {
  double payload = 1; // Ex: 20 (graus)
  string id = 2;
}

message UpdateStatus {
  string eventType = 9; // Ex: temperature
  double payload = 10; // Ex: 20 (graus)
}
